### ISSUE TITLE
Font Library: Fix vertical alignment of font face preview

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58439]
### Description

In the Fonts modal dialog, the vertical alignment of checkboxes and font names looks ok when the font preview is text:

![font preview text](https://github.com/WordPress/gutenberg/assets/1682452/278540ef-ed0e-4b84-8149-921a78e1cb00)

it doesn't when the font previs is an image:

![font preview image](https://github.com/WordPress/gutenberg/assets/1682452/eea9aade-eab7-4e5d-9443-1c1b6928d1b4)

This isn't surprising, as when an image `display` CSS property is the default `inline` value, images do reserve some 'descender space' to align with potential text that is placed in line with the image:

![descender space](https://github.com/WordPress/gutenberg/assets/1682452/58423603-1e03-4710-9014-c656edbf0626)

These kind of vertical alignments should be easily handled with CSS flexbox. Actually, the font 'row' does use flexbox but the font image (or text) is actually wrapped within an additional `div` elements that is the actual flex item, thus preventing flexbox to do its job on the image.

The [additional div element](https://github.com/WordPress/gutenberg/blob/6a84d8f831ec61b7953ad6296ce444b696ade239/packages/edit-site/src/components/global-styles/font-library-modal/font-demo.js#L68) is used as a reference (ref) for an `window.IntersectionObserver` instance so I'm not sure it can be removed.

If it can't be removed, the old-style solution would ne setting `display: block` on the image to remove the descender space and imrpove the vertical alignment.

Also, the height of the text preview and the one of the image preview is different. 22 pixels vs. 23 pixels:

![font preview text height](https://github.com/WordPress/gutenberg/assets/1682452/53b5af68-1d92-49bd-bbbd-6de2db958154)

![font preview image height](https://github.com/WordPress/gutenberg/assets/1682452/126a6198-2904-4520-bee0-73bfdc9ad0a3)

I'm not sure why the image preview is set to 23 pixels, that seems to me a 'magic number' that has no rationale. Also, the image has inline styling, see:

https://github.com/WordPress/gutenberg/blob/6a84d8f831ec61b7953ad6296ce444b696ade239/packages/edit-site/src/components/global-styles/font-library-modal/font-demo.js#L42-L45

Inline style should be avoided and a CSS class name should be used instead, to make possible to use .scss variables in the appropriate file.

### Step-by-step reproduction instructions

- Go to Site editor > Styles > Typography > Manage fonts.
- Click on a font that has an image-based preview e.g. the Google Fonts > Actor.
- Observe the alignment of the checkbox and image is off. 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
As an additional CSS issue, the opacity transition set on the textual demo doesn;'t respect the reduce motion preference.
